### PR TITLE
Add CoC

### PR DESCRIPTION
**Pull request by AtiusAmy**
Added Code of Conduct (GNOME's one to be more specific)

### CODE REVIEW

**Comment by runarcn** [comment]
Personally, I find a CoC sorta overkill for such a small project, but I get why people want it. I'm not gonna argue for or against as that's @heliguy4599's decision, but I think this one needs some improvements.

Those who believe someone breaks CoC shouldn't have to go into a public discord or matrix room to report it. I suggest adding an option to contact via mail - and make this the main way of doing it, as requiring more accounts and more platforms is just fragmenting. Every GitHub user needs an email anyways.

The CoC in it's entirety should also be added to the .md-file instead of only link to an external website. It'd be stupid if GNOME website happened to be down when someone checks, or if someone has crap internet and can't read it in it's entirety after cloning the repo with git. There's also the fact that the GNOME CoC is hosted on several places and could be removed from this link without anyone on the Warehouse project noticing for an extended amount of time.

It's also weird to link to a CoC that asks you to contact the GNOME CoC Comittee while stating here that you should contact over Discord/Matrix. It also specifies a scope which in no way includes the Warehouse git repository, unless this has become a GNOME Community Space without my knowledge.

All in all, I think this needs more polishing if this is to be merged.

**Comment by runarcn** [comment]
Apparently, we've gotten some attention from The Linux Experiment and omgubuntu among others. I take back my statement of "we don't need a coc since we're so small" as it seems like that could change.

I still believe we should (1) make people report via email (or matrix for encrypted communication), and (2) have a coc in plaintext instead of a hyperlink to the GNOME website

**Comment by runarcn** [comment]
I've been thinking some more about it, and think we should discuss this some more before it's merged. There are some things which need to be clarified before we can add a Code of Conduct. If we base ourself on the GNOME CoC, these are:

1. How we are to handle reports
2. Who should be allowed to handle reports
3. How do we set up a sort of modmail, and how do we make it sufficiently encrypted and privacy respecting

Before we have proper answers to these three, we can't really add a CoC.

**Comment by AtiusAmy** [comment]
Yeah, we should create some kind of modmail.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,3 +1,168 @@
...
 If you believe that someone is violating the Code of Conduct, or have any other concerns, please contact us via [Discord](https://discord.gg/Sq85C42Xkt) or [Matrix](https://matrix.to/#/#warehouse:beeper.com).
+
+The GNOME Code of Conduct is also included at the end of this file.
+
+
+GNOME Code of Conduct
+
+Thank you for being a part of the GNOME project. We value your participation and want everyone to have an enjoyable and fulfilling experience. Accordingly, all participants are expected to follow this Code of Conduct, and to show respect, understanding, and consideration to one another. Thank you for helping make this a welcoming, friendly community for everyone.
+
+Scope
```
Comment (runarcn) [inline CODE_OF_CONDUCT.md:12]: unecessary to define the entire gnome scope

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -1,3 +1,168 @@
 The project follows the [GNOME Code of Conduct](https://wiki.gnome.org/Foundation/CodeOfConduct).
 
 If you believe that someone is violating the Code of Conduct, or have any other concerns, please contact us via [Discord](https://discord.gg/Sq85C42Xkt) or [Matrix](https://matrix.to/#/#warehouse:beeper.com).
+
+The GNOME Code of Conduct is also included at the end of this file.
+
+
+GNOME Code of Conduct
+
+Thank you for being a part of the GNOME project. We value your participation and want everyone to have an enjoyable and fulfilling experience. Accordingly, all participants are expected to follow this Code of Conduct, and to show respect, understanding, and consideration to one another. Thank you for helping make this a welcoming, friendly community for everyone.
```
Comment (runarcn) [inline CODE_OF_CONDUCT.md:10]: consider changing to "Thank you for being a part of the Warehouse project" and do that elsewhere as well?

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -1,3 +1,168 @@
...
+
+All participants in GNOME community spaces are subject to the Code of Conduct. This includes GNOME Foundation board members, corporate sponsors, and paid employees. This also includes volunteers, maintainers, leaders, contributors, contribution reviewers, issue reporters, GNOME users, and anyone participating in discussion in GNOME community spaces. For in-person events, this also includes all attendees, exhibitors, vendors, speakers, panelists, organizers, staff, and volunteers.
+
+Reporting an Incident
+
+If you believe that someone is violating the Code of Conduct, or have any other concerns, please contact the Code of Conduct committee.
```
Comment (runarcn) [inline CODE_OF_CONDUCT.md:36]: change to "please contact us via [email](mailto:heli@guy.4577). Alternatively, use [Matrix](link.tld)" or similar?

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -1,3 +1,168 @@
...
+
+Outreach and diversity efforts directed at under-represented groups are permitted under the code of conduct. For example, a social event for women would not be classified as being outside the Code of Conduct under this provision.
+
+Basic expectations for conduct are not covered by the "reverse-ism clause" and would be enforced irrespective of the demographics of those involved. For example, racial discrimination will not be tolerated, irrespective of the race of those involved. Nor would unwanted sexual attention be tolerated, whatever someone's gender or sexual orientation. Members of our community have the right to expect that participants in the project will uphold these standards.
+
+If a participant engages in behavior that violates this code of conduct, the GNOME Code of Conduct committee may take any action they deem appropriate. Examples of consequences are outlined in the Committee Procedures Guide.
+
+Procedure for Handling Incidents
+
+    Reporter Guide
```
Comment (runarcn) [inline CODE_OF_CONDUCT.md:126]: make sure to add the links using markdown

**Comment by heliguy4599** [comment]
This has been closed as #45 exists as an updated PR for this issue

**Pull request closed without merging**